pkg/process/checks/discovery_check.go

```diff
@@ -0,0 +1,63 @@
+package checks
+
+import (
+	"os"
+
+	model "github.com/DataDog/agent-payload/process"
+	"github.com/DataDog/datadog-agent/pkg/process/config"
+	"github.com/DataDog/datadog-agent/pkg/process/procutil"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+var Discovery = &DiscoveryCheck{}
+
+type DiscoveryCheck struct {
```

Comment:
Mind renaming it to `ProcessDiscoveryCheck` - there are other components in the process agent unrelated to process collection.